src/midiEditorCommands.cpp

```diff
@@ -404,7 +498,6 @@ const string getMidiNoteName(MediaItem_Take *take, int pitch, int channel) {
 	}
 	return s.str();
 }
-
 // Returns the indexes of the first and last notes in a chord in a given direction.
 pair<int, int> findChord(MediaItem_Take* take, int direction) {
```

Comment:
I'm wondering whether this should eventually return a pair of MidiNoteIterators. I guess that'd require passing in ReqParams in case the caller wants something else. And there's the challenge of what to return instead of -1. We can deal with that some other time, though.

Reply:
I thought about that too when having lunch today, haha. Point is that for the chords, we just need the start for the note, whereas for preview, we need much more parameters from REAPER (vel, chan, etc.)

Having said that, I wonder whether REAPER's `MIDI_GetNote` performance is affected by requesting more parameters. Most likely it is just getting a note and writing values that are already in memory to the provided output variables. That doesn't apply to start and end, though.

Reply:
Sure, but the function could take ReqParams and we could pass in different ReqParams depending on whether we need preview.